.pre-commit-config.yaml

```diff
@@ -0,0 +1,41 @@
+repos:
+  - repo: https://github.com/PyCQA/flake8
+    rev: 6.0.0
+    hooks:
+      - id: flake8
+        additional_dependencies:
+          - flake8-executable
+          - bandit
+          - flake8-bugbear
+          - flake8-comprehensions
+          - flake8-simplify
+        args: [ "--max-line-length", "120" ]
+        exclude: "tests/to-be-tested/"
+  - repo: https://github.com/pre-commit/mirrors-mypy
+    rev: 'v0.991'
+    hooks:
+      - id: mypy
+        args: ["--disallow-untyped-defs", "--show-error-codes"]
+        exclude: "tests/"
+  - repo: https://github.com/psf/black
+    rev: 22.12.0
+    hooks:
+      - id: black
+        exclude: "tests/to-be-tested/"
+  - repo: https://github.com/pycqa/isort
+    rev: 5.11.4
+    hooks:
+      - id: isort
+        args: [ "--profile", "black", "-w", "88" ]
```

Comment:
Why "88"?

Reply:
Got some compatibility issues, but following up the chat below I've reduced the set of changes to not include additional tools (e.g. isort, black, pyupgrade) to speed up things.